src/error.rs

```diff
@@ -58,11 +60,38 @@ impl Error {
     #[inline]
     pub fn is_redirect(&self) -> bool {
         match self.kind {
-            Kind::TooManyRedirects |
-            Kind::RedirectLoop => true,
+            Kind::TooManyRedirects | Kind::RedirectLoop => true,
+            _ => false,
+        }
+    }
+
+    /// Returns true if the error is from a request returning a 4xx error.
+    #[inline]
+    pub fn is_client_error(&self) -> bool {
```

Comment:
I wonder if there'd be confusing that this means a `status.is_client_error()`, and not an error from `reqwest::Client`.